src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedParameterSymbol.cs

```diff
@@ -213,12 +217,13 @@ internal sealed class SynthesizedParameterSymbol : SynthesizedParameterSymbolBas
             RefKind refKind,
             string name = "",
             DeclarationScope scope = DeclarationScope.Unscoped,
+            ConstantValue? defaultValue = null,
             ImmutableArray<CustomModifier> refCustomModifiers = default,
             SourceComplexParameterSymbolBase? baseParameterForAttributes = null)
         {
             if (refCustomModifiers.IsDefaultOrEmpty && baseParameterForAttributes is null)
             {
-                return new SynthesizedParameterSymbol(container, type, ordinal, refKind, scope, name);
+                return new SynthesizedParameterSymbol(container, type, ordinal, refKind, scope, defaultValue, name);
```

Comment:
Yes this seems reasonable to me, in which case it might be worth renaming `SynthesizedParameterSymbol` to `SynthesizedSimpleParameterSymbol`. I've made that change

Reply:
On second thought, perhaps renaming wouldn't be the clearest, due to the presence of the `Create` factory method.

Reply:
@adamperlin The static method is probably better on `SynthesizedParameterSymbolBase`?

@333fred What are your thoughts here?

Reply:
What method are you referring to @Youssef1313?

Reply:
`Create` factory method

https://github.com/dotnet/roslyn/blob/0f757da6204418a21dbf129b90831f5200a2e388/src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedParameterSymbol.cs#L213-L239

Reply:
We didn't create that method here, so let's not move it as part of this pr.

Reply:
@Youssef1313 yes I think it makes more sense in `SynthesizedParameterSymbolBase`, but I won't move it in this pr.